docs/development/manual_tests.md

```diff
@@ -80,6 +80,54 @@ Tool Specific combinations
+
+## Test: (3) Two reference FASTAs, one also has dict ✅
+## Expect: Expect two fai file (x2 SAMTOOLS_FAIDX processes), 1 dict file (x1 PICARD_CREATESEQUENCEDICTIONARY), 2 bwa index directory (x2 BWA_INDEX) etc. to be generated and present in per reference results/reference folder
+nextflow run ../main.nf -profile singularity,test --outdir ./results --input samplesheet.tsv --fasta reference_sheet_multiref_test03.csv -ansi-log false -dump-channels --save_reference
+
+## Test: (4) Two reference FASTAs, one also has bwa_index ref ✅
+## Expect: Expect two fai (x2 SAMTOOLS_FAIDX processes), two dict (x2 PICARD_CREATESEQUENCEDICTIONARY), 1 bwa  index directory (x1 BWA_INDEX) etc. to be generated and present in per reference results/reference folder
+nextflow run ../main.nf -profile singularity,test --outdir ./results --input samplesheet.tsv --fasta reference_sheet_multiref_test04.csv -ansi-log false -dump-channels --save_reference
+
+## Test: (5) Two reference FASTAs, one also has bowtie2_index ref ✅
+## Expect: Expect two fai (x2 SAMTOOLS_FAIDX processes), two dict (x2 PICARD_CREATESEQUENCEDICTIONARY), 1 bowtie2  index directory (x1 BOWTIE2_BUILD) etc. to be generated and present in per reference results/reference folder
+nextflow run ../main.nf -profile singularity,test --outdir ./results --input samplesheet.tsv --fasta reference_sheet_multiref_test05.csv -ansi-log false -dump-channels --save_reference --mapping_tool bowtie2
+
+## Test: (6) Mapper index mismatch with `--mapping_tool` ✅
+## Expect: Expect FAIL at mapping step for Mammoth two fai (x2 SAMTOOLS_FAIDX processes), two dict (x2 PICARD_CREATESEQUENCEDICTIONARY), 1 bowtie2  index directory (BOWTIE2_BUILD) etc. to be generated and present in per reference results/reference folder
+nextflow run ../main.nf -profile singularity,test --outdir ./results --input samplesheet.tsv --fasta reference_sheet_multiref_test06.csv -ansi-log false -dump-channels --save_reference --mapping_tool bowtie2
+
+## Test: (7) Mammoth has all pre-supplied ✅
+## Expect: Expect one fai (x1 SAMTOOLS_FAIDX processes), one dict (x1 PICARD_CREATESEQUENCEDICTIONARY), 1 bowtie2  index directory (BOWTIE2_BUILD) etc. to be generated and present in per reference results/reference folder
+nextflow run ../main.nf -profile singularity,test --outdir ./results --input samplesheet.tsv --fasta reference_sheet_multiref_test07.csv -ansi-log false -dump-channels --save_reference
+
+## Test: (8) No indexing necessary, all already supplied ✅
+## Expect: Expect no files to be generated/processes executed, nor results present in per reference results/reference folder
+nextflow run ../main.nf -profile singularity,test --outdir ./results --input samplesheet.tsv --fasta reference_sheet_multiref_test08.csv -ansi-log false -dump-channels --save_reference
+
+## Test: (9) All but Human FAI provided ❓️ - fails at step not involving FAI?
```

Comment:
NO ✅ ?

Reply:
Ah completely forgot about that

```
Execution completed unsuccessfully!
The exit status of the task that caused the workflow execution to fail was: 1.

The full error message was:

Error executing process > 'NFCORE_EAGER:EAGER:MAP:FASTQ_ALIGN_BWAALN:BWA_SAMSE (hs37d5_chr21|JK2802_JK2802_AGAATAACCTACCA_L2)'

Caused by:
  Process `NFCORE_EAGER:EAGER:MAP:FASTQ_ALIGN_BWAALN:BWA_SAMSE (hs37d5_chr21|JK2802_JK2802_AGAATAACCTACCA_L2)` terminated with an error exit status (1)

Command executed:

  INDEX=`find -L ./ -name "*.amb" | sed 's/\.amb$//'`
  
  bwa samse \
      -r '@RG\tID:ILLUMINA-JK2802_AGAATAACCTACCA\tSM:JK2802\tPL:illumina\tPU:ILLUMINA-JK2802_AGAATAACCTACCA-double' \
       \
      $INDEX \
      JK2802_JK2802_AGAATAACCTACCA_hs37d5_chr21.sai \
      JK2802_JK2802_AGAATAACCTACCA_L2_JK2802_AGAATAACCTACCA.fastp.fastq.gz | samtools sort -@ 1 -O bam - > JK2802_JK2802_AGAATAACCTACCA_hs37d5_chr21.bam
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_EAGER:EAGER:MAP:FASTQ_ALIGN_BWAALN:BWA_SAMSE":
      bwa: $(echo $(bwa 2>&1) | sed 's/^.*Version: //; s/Contact:.*$//')
      samtools: $(echo $(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*$//')
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)

Command error:
  INFO:    Environment variable SINGULARITYENV_NXF_DEBUG is set, but APPTAINERENV_NXF_DEBUG is preferred
  WARNING: Skipping mount /var/lib/apptainer/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container
  [bwa_aln_core] convert to sequence coordinate... samtools sort: failed to read header from "-"

Work dir:
  /home/james/git/nf-core/eager/testing/work/b9/d46fcf874dd9c0663a96d4df205133

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`
```
Investigating...

Reply:
OK I have no clue... will need to rubberduck with you @TCLamnidis

Reply:
The _actual_ error when removing the pipe

```
[bwa_aln_core] convert to sequence coordinate... /home/james/git/nf-core/eager/testing/work/b9/d46fcf874dd9c0663a96d4df205133/.command.sh: line 9:    43 Segmentation fault      (core dumped) bwa samse -r '@RG\tID:ILLUMINA-JK2802_AGAATAACCTACCA\tSM:JK2802\tPL:illumina\tPU:ILLUMINA-JK2802_AGAATAACCTACCA-double' $INDEX JK2802_JK2802_AGAATAACCTACCA_hs37d5_chr21.sai JK2802_JK2802_AGAATAACCTACCA_L2_JK2802_AGAATAACCTACCA.fastp.fastq.gz > what.bam
```